src/sankoff_handler.cpp

```diff
@@ -3,103 +3,158 @@
-            for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount(); pattern_idx++) {
-                Assert(node->Children().size() == child_count, "Error in SankoffHandler::RunSankoff: Tree should be bifurcating.");
-                // Which child partial is in right or left doesn't actually matter because they are summed when calculating q_partials. In DAG case, 
-                // will want to ensure that left children partials are in p_partials_left
-                p_partials_left[pattern_idx].block<1, state_count>(node->Id(), 0) = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[0]->Id()));
-                p_partials_right[pattern_idx].block<1, state_count>(node->Id(), 0) = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[1]->Id()));
-            }
-        }
-    });
+SankoffPartialVec SankoffHandler::BlankPartials(size_t node_count) {
+  SankoffPartialVec blank_partials(site_pattern_.PatternCount());
+  for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
+       pattern_idx++) {
+    SankoffPartial blank_partials_per_site(node_count, state_count);
+    blank_partials_per_site.setZero();
+    blank_partials[pattern_idx] = blank_partials_per_site;
+  }
+  return blank_partials;
+}
 
-    // generating q-partials
-    auto q_partials = BlankPartials(topology->Id()+1);
-    topology->Preorder([&q_partials, &p_partials_left, &p_partials_right, this](const Node* node) {
-        if (!node->IsLeaf()) {
-            Assert(node->Children().size() == child_count, "Error in SankoffHandler::RunSankoff: Tree should be bifurcating.");
-            for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount(); pattern_idx++) {
-                auto partials_from_parent = ParentPartial(q_partials[pattern_idx].block<1, state_count>(node->Id(), 0));
-                for (size_t child = 0; child < child_count; child++) {
-                    size_t sister_idx = child == 0 ? 1 : 0;
-                    auto partials_from_sister = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[sister_idx]->Id()));
-                    q_partials[pattern_idx].block<1, state_count>(node->Children()[child]->Id(), 0) = partials_from_sister + partials_from_parent;
-                }
-            }
-        } 
-    });
+SankoffPartialVec SankoffHandler::RunSankoff(Node::NodePtr topology) {
+  // generating p_partials (right and left)
+  SankoffPartialVec p_partials_left = GenerateLeafPartials(topology->Id() + 1);
+  auto p_partials_right = BlankPartials(topology->Id() + 1);
+  topology->Postorder([&p_partials_left, &p_partials_right, this](const Node* node) {
+    if (!node->IsLeaf()) {
+      for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
+           pattern_idx++) {
+        Assert(node->Children().size() == child_count,
```

Comment:
`child_count` is a variable and the error is about exactly 2.

Reply:
@matsen I don't understand this comment. child_count is to assert that the tree is bifurcating.

Reply:
Right-- I'm saying that it makes more sense to just write 2 because that's what the error message in the assert says. I'm not sure if it makes sense to have child_count at all-- I can't imagine a setting in which we would want a value other than 2.

Reply:
(But I appreciate the intent to keep things clear and tidy!)